metatron/src/main/java/uk/gov/ida/eidas/metatron/health/MetadataHealth.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.ida.eidas.metatron.health;
+
+import org.joda.time.DateTime;
+import org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver;
+
+public class MetadataHealth {
+
+    private String entityId;
+    private DateTime expirationTime;
+    private DateTime lastRefresh;
+    private DateTime lastSuccessfulRefresh;
+    private boolean wasLastRefreshSuccessful;
+    private DateTime nextRefresh;
+
+    public MetadataHealth(String entityId, AbstractReloadingMetadataResolver resolver) {
+        this.entityId = entityId;
```

Comment:
Rather than having all the elements as separate fields, could you just have `entityId` and `resolver`, and then have the getters call methods on the resolver? I know that at the moment the methods are being called immediately after the object is created, so it doesn't matter much, but if this was ever used somewhere else where the object is created a while before it's accessed, you might end up with stale data.

Reply:
I don't expect this to live too long.  It's useful for poking and meeting the original acceptance criteria but I'm pretty sure it's going to be superseded by Prometheus which will work completely differently.

Having a reference to a resolver would make more sense but I suspect that once we start retrieving actual metadata data there will be an extra layer of abstraction that will pretty much do what you're saying.